workspaces/mi/mi-extension/src/rpc-managers/mi-diagram/rpc-manager.ts

```diff
@@ -4273,48 +4278,59 @@ ${endpointAttributes}
-                            }
-                        } else {
-                            legacyConnectorCache.set('inbound-connector-data', data['inbound-connector-data']);
-                            legacyConnectorCache.set('outbound-connector-data', data['outbound-connector-data']);
-                            if (connectorStoreData) {
-                                legacyConnectorCache.set('connectors', connectorStoreData);
-                            }
+            const response = await fetch(APIS.MI_CONNECTOR_STORE);
+            const connectorStoreResponse = await fetch(
+                APIS.MI_CONNECTOR_STORE_BACKEND.replace('${version}', runtimeVersion ?? '')
+            );
+
+            const data = await response.json();
+            const connectorStoreData = await connectorStoreResponse.json();
+
+            if (data && data['inbound-connector-data'] && data['outbound-connector-data']) {
+                if (runtimeVersion) {
+                    if (compareVersions(runtimeVersion, RUNTIME_VERSION_440) >= 0) {
+                        connectorCache.set('inbound-connector-data', data['inbound-connector-data']);
+                        connectorCache.set('outbound-connector-data', data['outbound-connector-data']);
+                        if (connectorStoreData) {
+                            connectorCache.set('connectors', connectorStoreData);
+                        }
+                    } else {
+                        legacyConnectorCache.set('inbound-connector-data', data['inbound-connector-data']);
+                        legacyConnectorCache.set('outbound-connector-data', data['outbound-connector-data']);
+                        if (connectorStoreData) {
+                            legacyConnectorCache.set('connectors', connectorStoreData);
                         }
                     }
-                    resolve({ inboundConnectors: data['inbound-connector-data'], outboundConnectors: data['outbound-connector-data'], connectors: connectorStoreData });
-                } else {
-                    console.log("Failed to fetch connectors. Status: " + data.status + ", Reason: " + data.reason);
-                    reject("Failed to fetch connectors.");
                 }
-            } catch (error) {
-                console.log("User is offline.", error);
-                reject("Failed to fetch connectors.");
-            }
-        });
 
+                return {
+                    inboundConnectors: data['inbound-connector-data'],
+                    outboundConnectors: data['outbound-connector-data'],
+                    connectors: connectorStoreData,
+                };
+            } else {
+                console.log("Failed to fetch connectors. Status: " + data.status + ", Reason: " + data.reason);
+                throw new Error("Failed to fetch connectors.");
+            }
+        } catch (error) {
+            console.log("User is offline.", error);
```

Comment:
We should show a warning to the user and give an option to skip library downloading to avoid any confusion.